### ISSUE TITLE
Audit: Sign in with GitHub

### ISSUE THREAD

**Issue by 0x4007** [ubiquity/audit.ubq.fi#3]
Incorporate a sign in with GitHub button for increased auth limits. Don't worry about styling, I can handle it. 

https://github.com/ubiquity/work.ubq.fi/blob/development/src/home/authentication.ts

> We should take the sign in with GitHub implementation I made on work.ubq.fi and authenticate as the user for increased rate limits. 

> How is authentication being handled now?

_Originally posted by @pavlovcik in https://github.com/ubiquity/pay.ubq.fi/pull/164#discussion_r1496920814_

**Comment by guha-rahul** [comment]
The sign in with Github is already implemented. Can u please explain a little more?

**Comment by 0x4007** [comment]
> The sign in with Github is already implemented. Can u please explain a little more?

There's no sign in with GitHub implemented on this project.

**Comment by molecula451** [comment]
if it's going to be refactor i think it's best to focus energy on ubiquity/pay.ubq.fi#174 and move all these tasks there

**Comment by 0x4007** [comment]
This is only needed in the audit app. So actually this issue will be transferred to that repository once its broken off from this one.

**Comment by gentlementlegen** [comment]
@pavlovcik this should be taken care of after eventually resolving https://github.com/ubiquity/pay.ubq.fi/pull/172
and actually, once this is merged, calls to GitHub API are extremely low since it would only query the repo information, and not all the comments / issues etc, making this issue low prio.

**Comment by 0x4007** [comment]
Priority 1 normal is our lowest priority!

**Comment by gentlementlegen** [comment]
Partially solved by https://github.com/ubiquity/audit.ubq.fi/issues/5 for the auth limits, since it would just poke the database and eventually GitHub a very little for repository informations.

**Comment by 0x4007** [comment]
> Partially solved by https://github.com/ubiquity/audit.ubq.fi/issues/5 for the auth limits, since it would just poke the database and eventually GitHub a very little for repository informations.

Cool to hear. Meaning it solves the rate limit issue but doesn't include the sign in with GitHub button? This should be fine to get some credit for since #5 is technically a different scope 

I also realize that the pricing didn't make it over. @molecula451 would you happen to have a record of the priority and time levels for the issues? Can you re add them so that compensation can be distributed?

**Comment by gentlementlegen** [comment]
> > Partially solved by #5 for the auth limits, since it would just poke the database and eventually GitHub a very little for repository informations.
> 
> Cool to hear. Meaning it solves the rate limit issue but doesn't include the sign in with GitHub button? This should be fine to get some credit for since #5 is technically a different scope
> 
> I also realize that the pricing didn't make it over. @molecula451 would you happen to have a record of the priority and time levels for the issues? Can you re add them so that compensation can be distributed?

Correct, because we don't require any login to access our database, and we have no rate limit. Now the ratio would be one hit to GitHub API per repo audited, so if you audit 5000 repositories within an hour you might hit the API restrictions. Could still be a nice to have, but not very likely to happen is it?

From the GitHub docs REST API
```
All of these requests count towards your personal rate limit of 5,000 requests per hour. Requests made on your behalf by a GitHub App that is owned by a GitHub Enterprise Cloud organization have a higher rate limit of 15,000 requests per hour.
```

**Comment by molecula451** [comment]
> I also realize that the pricing didn't make it over. @molecula451 would you happen to have a record of the priority and time levels for the issues? Can you re add them so that compensation can be distributed?

Yes, this was the actual price label:


![Screenshot from 2024-03-04 23-29-43](https://github.com/ubiquity/audit.ubq.fi/assets/41552663/c3794e59-c405-489b-92a7-db9fd440d48c)

**Comment by molecula451** [comment]
i can bump if anything, tho
